### ISSUE TITLE
Publish on conda-forge and nixpkgs

### ISSUE THREAD

**Issue by alippai** [microsoft/pyright#2076]
**Is your feature request related to a problem? Please describe.**
Python projects often use conda forge and nix repositories as a primary source for build tools. Publishing pyright on those platforms would help a lot with adoption, make the beginners life easier and could provide a better ops and developer experience overall.

**Comment by erictraut** [comment]
Installing pyright locally requires that node is installed. Once node is installed, installation is as easy as `npm install pyright`.

If you're installing pyright as part of your continuous integration (CI) system, you can use a [github action](https://github.com/jakebailey/pyright-action) to automate the install.

So, I think it's already straightforward to install pyright. I don't think that we can make it any easier by publishing it to more locations.

We publish pyright frequently (typically twice a week), so publishing to more places takes extra time, bandwidth, and storage. I'm reluctant to expand the list of publish locations without any clear benefit.

**Comment by alippai** [comment]
Yes, installing Node or Docker, using GitHub Actions can sound easy, however it results in an additional toolchain detached from the existing python environments. A lot of CLI tools (being rust, node or python based) deploy their binaries on those platforms. Both platforms make it easy to build and configure once and reuse the built binaries and configurations on multiple platform - no need to tackle issues with proxies, incompatible glibc versions, SSL CAs. 

In short: we could use only a single tool instead of maintaining two.

I understand this might not be your priority, in this case maybe a community effort will work instead. As projects (being MS owned or OSS) tend to keep control the publishing pipeline of their packages, I thought it's better to propose it first here.

**Comment by RobertCraigie** [comment]
@alippai I recently released a package on [PyPI](https://pypi.org/project/pyright/) to support installing pyright with [pip](https://pip.pypa.io/en/stable/) and while the solution I have implemented is far from ideal (node will be downloaded at runtime if not present) it should still work, however I would not use this for any mission-critical project just yet.

I don't have any knowledge on conda or nix repositories so I can't comment on whether or not this will work for your use case.

You can find the source code [here](https://github.com/RobertCraigie/pyright-python)

**Comment by alippai** [comment]
@RobertCraigie thanks!. Packages in conda and nix can depend on nodejs (with specific version, automatically loaded into path). This is why they are popular. It's not "nodejs only" or "python only", but batteries included. Imagine you'd need to install specific C++/GPU libs to work with TensorFlow or pytorch...

**Comment by erictraut** [comment]
@RobertCraigie, I like the approach you've taken. Others have proposed bundling the node binary and a static snapshot of pyright in the package, which I don't like. But you're dynamically installing node and pyright is preferable to bundling them in the package. This solves a number of technical and licensing issues. My only concern is that it is pinned to a specific version of pyright, which will inevitably get out of date. It's already 6 versions old. I don't want to be fielding bug reports from users who are using old versions of pyright. Perhaps you would consider modifying your script to download the latest version by default rather than a hard-coded version? You could crib from [this shell script](https://github.com/microsoft/pyright/blob/main/docs/ci-integration.md#running-pyright-from-a-ci-script) that queries for the latest pyright version.

@alippai, if you want to create a similar wrapper in other tools registries, I'm not opposed to that. We currently have no plans to do so.

**Comment by alippai** [comment]
@erictraut superb, thanks!

**Comment by RobertCraigie** [comment]
@erictraut Thank you!

Downloading the latest version is a good suggestion, I will look into adding that as soon as possible.

**Comment by RobertCraigie** [comment]
@erictraut Automatic version checks added in [v0.0.4](https://github.com/RobertCraigie/pyright-python/commit/294c5d85dc8565dee76b5111776fc413d219a5b8)

**Comment by erictraut** [comment]
@RobertCraigie, that great. Thanks!

**Comment by RobertCraigie** [comment]
@alippai My https://github.com/RobertCraigie/pyright-python package has now been published on `conda-forge`!

Should be noted that I have not contributed to or tested it but the package can be found here: https://github.com/conda-forge/pyright-feedstock

@erictraut Would it be worth mentioning these community maintained packages in the documentation?
